anitya/lib/backends/maven.py

```diff
@@ -70,3 +75,96 @@ def get_versions(cls, project):
+        temp_artifact = do_replace(temp_artifact, match_non_decimal, match_non_decimal)
+
+        # do same thing with group id
+        temp_group = do_replace(group_id, match_decimal, match_non_decimal)
+        temp_group = do_replace(temp_group, match_non_decimal, match_decimal)
+        temp_group = do_replace(temp_group, match_non_decimal, match_non_decimal)
+
+        homepage = do_request(temp_group, temp_artifact)
+        if homepage is not None:
+            return homepage
+        homepage = do_request(temp_group, artifact_id)
+        if homepage is not None:
+            return homepage
+        homepage = do_request(group_id, temp_artifact)
+        if homepage is not None:
+            return homepage
+        homepage = do_request(group_id, artifact_id)
+        if homepage is not None:
+            return homepage
+
+    @classmethod
+    def check_feed(cls):
+        '''check_feed for Maven backend.
+
+        Return a generator over the latest 40 packages to Maven central index by
+        calling maven-index-checker application
+
+        Returns:
+            generator over new packages
+        '''
+        if APP.config['JAVA_PATH'] is None:
+            raise AnityaPluginException('no java binary specified')
+        if APP.config['JAR_NAME'] is None:
+            raise AnityaPluginException('no maven-release-checker jar file specified')
+
+        try:
+            data = check_output([APP.config['JAVA_PATH'], "-jar",
+                                 APP.config['JAR_NAME'], "-it"], universal_newlines=True)
+        except CalledProcessError:
+            raise AnityaPluginException(
+                'maven-release-checker exited with non zero value')
+        for item in json.loads(data):
+            maven_coordinates = '{group_id}:{artifact_id}'. \
+                format(group_id=item['groupId'], artifact_id=item['artifactId'])
+            try:
+                projects = Project.query.filter(Project.version_url
+                                                == maven_coordinates).all()
+            except NoResultFound:
+                projects = []
+
+            if len(projects) != 0:
+                # If there is project created it can have
+                # different name than maven_coordinates
+                name = projects[0].name
+                homepage = projects[0].homepage
+            else:
+                name = maven_coordinates
+                homepage = cls.verify_url(item['groupId'], item['artifactId'])
+                if homepage is None:
+                    continue
+            version = item['version']
+            yield name, homepage, cls.name, version
+
+
+def do_replace(input_string, first_pattern, second_pattern):
+    return re.sub(r'{first_pattern}\.{second_pattern}'.
+                  format(first_pattern=first_pattern,
+                         second_pattern=second_pattern), r'/', input_string)
+
+
+def do_request(group_id, artifact_id):
```

Comment:
I think you can rewrite the code that uses this function to only need to call this once, so this function can go away as well.